### PR TITLE
New topic: CVEs (various)

### PR DESCRIPTION
**Pull request by harding**

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

Very nice work! Thanks!

You might want to double check against this list https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Fuzz-Trophies to make sure no CVE assignments are missing.

**Comment by harding** [comment]
@practicalswift 

> You might want to double check against this list bitcoin-core/bitcoin-devwiki/wiki/Fuzz-Trophies to make sure no CVE assignments are missing.

I'm sure there's several notable CVEs missing since this list is just for CVEs mentioned in Optech content.  This list was produced using boring old `git grep CVE`.  The page does link to https://en.bitcoin.it/wiki/Common_Vulnerabilities_and_Exposures#CVE-2018-17145 as its primary source, as that's the most comprehensive list of Bitcoin and LN CVEs of which I'm aware.

File: _topics/en/cve.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        timely manner and so allowed funds to be stolen.  Fixed in LND
+        0.10.0-beta
+
+    CVE-2018-17145:
+      link: /en/newsletters/2020/09/16/#inventory-out-of-memory-denial-of-service-attack-invdos
+      summary: >
+        Flooding a node with an excess of `inv` messages could lead to
+        the node running out of memory and crashing, increasing the risk
+        of [eclipse attacks][topic eclipse attacks] that can steal
+        money.  Affected Bitcoin Core, btcd, bcoin, and multiple
```
Comment (bitschmidty) [inline _topics/en/cve.md:113]: Inconsistent casing in node implementation naming

File: _topics/en/cve.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        in response to foreign routing requests, which could've allowed
+        funds to be stolen.  Fixed in LND 0.11.0-beta.
+
+    CVE-2020-26895:
+      link: /en/newsletters/2020/10/28/#cve-2020-26895-acceptance-of-non-standard-signatures
+      summary: >
+        Vulnerability in LND due to accepting non-standard "high-S"
+        signatures, which could've prevented channels from closing in a
+        timely manner and so allowed funds to be stolen.  Fixed in LND
+        0.10.0-beta
```
Comment (bitschmidty) [inline _topics/en/cve.md:105]: s/0.10.0-beta/0.10.0-beta.

File: _topics/en/cve.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+---
+Some CVEs that have been frequently mentioned by Optech have their own
+pages in the Optech topic index (see this page's *see also* section for
+links).  For any other CVEs affecting Bitcoin and Lightning, this
+catch-all page collects any significant mentions of those CVEs in Optech
+content.  This pages's CVE's include:
+
+{% for alias in page.aliases %}
+  * **[{{alias}}]({{page.extra.cves[alias].link | default: "#CVE-LINK-NOT-PROVIDED"}}):** {{page.extra.cves[alias].summary}}
```
Comment (bitschmidty) [inline _topics/en/cve.md:161]: Right now CVE-2021-31876's entry resolves to link not provided, perhaps it should just go to /en/newsletters/2021/05/12/#cve-2021-31876-discrepancy-between-bip125-and-bitcoin-core-implementation

File: _topics/en/cve.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    {%- if mention.title contains alias %}
+      * [{{mention.title}}]({{mention.url}})
+    {%- endif -%}
+  {%- endfor -%}<br><br>
+
+{% endfor %}
+
+Please note that CVE entries may be removed from this page if we end up
+creating a new topic page specifically for that CVE.  Any moved CVEs
+should be listed in the *see also* section below.
```
Comment (jnewbery) [inline _topics/en/cve.md:183]: What do you think about also listing them here, with just a link to their own topic page? That's slightly more convenient for readers than having to look in two places (and would also allow you to remove this note and the similar "Some CVEs that have been frequently mentioned by Optech have their own pages in the Optech topic index (see this page's *see also* section for
links)." note at the top of the page.

File: _topics/en/cve.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    CVE-2017-18350:
+      link: /en/newsletters/2019/11/13/#cve-2017-18350-socks-proxy-vulnerability
+      summary: >
+        A vulnerability in Bitcoin Core affecting users of SOCKS
+        proxies.  Fixed in Bitcoin Core 0.16.0.
+
+    CVE-2013-2292:
+      link: https://bitcointalk.org/?topic=140078
+      summary: >
+        A vulnerability in Bitcoin Core that it was believed could've
```
Comment (jnewbery) [inline _topics/en/cve.md:136]: I'm normally fine with contractions, but for some reason "could've" seems too informal. I know that's inconsistent and probably just a personal thing, so feel free to ignore.

File: _topics/en/cve.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+      summary: >
+        Vulnerability in LND due to improperly revealing local preimages
+        in response to foreign routing requests, which could've allowed
+        funds to be stolen.  Fixed in LND 0.11.0-beta.
+
+    CVE-2020-26895:
+      link: /en/newsletters/2020/10/28/#cve-2020-26895-acceptance-of-non-standard-signatures
+      summary: >
+        Vulnerability in LND due to accepting non-standard "high-S"
+        signatures, which could've prevented channels from closing in a
```
Comment (jnewbery) [inline _topics/en/cve.md:103]: Same comment about could've. Again, feel free to ignore if you disagree.

File: _topics/en/cve.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+        Bitcoin Core saw such an invalid block, they cached its
+        rejection and so would later reject the valid version of the
+        block, which could easily lead to short-term network partitions
+        where users could be tricked into accepting invalid bitcoins.
+        Fixed in Bitcoin 0.6.1.
+
+    CVE-2021-31876:
+      link: /en/newsletters/2021/05/12/#cve-2021-31876-discrepancy-between-bip125-and-bitcoin-core-implementation
+      summary: >
+        The BIP125 specification of opt-in Replace By Fee (RBF) says
```
Comment (jnewbery) [inline _topics/en/cve.md:156]: Maybe link to https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki

File: _topics/en/cve.md (hunk 5 above)
Comment (harding) [reply _topics/en/cve.md:136]: Nah, my spell checker doesn't think they're real contractions either; they're just such a part of my causal writing and speech patterns that I forget they're not normal for the seven billion people who aren't Piney hicks.  I'll expand the term.

**Comment by harding** [comment]
Updated, I think all feedback has been addressed.

**Comment by jnewbery** [comment]
Looks great. Thanks Dave!

**Pull request merged**